numba/unicode_support.py

```diff
@@ -251,15 +251,31 @@ def _PyUnicode_ToLowercase(ch):
 
 @register_jitable
 def _PyUnicode_ToLowerFull(ch, res):
-    raise NotImplementedError
+    ctype = _PyUnicode_gettyperecord(ch)
+    if (ctype.flags & _PyUnicode_TyperecordMasks.EXTENDED_CASE_MASK):
+        index = ctype.lower & 0xFFFF
+        n = ctype.lower >> 24
+        for i in range(n):
+            res[i] = _PyUnicode_ExtendedCase(index + i)
+        return n
+    res[0] = ch + ctype.lower
+    return 1
 
 
 @register_jitable
 def _PyUnicode_ToTitleFull(ch, res):
-    raise NotImplementedError
+    ctype = _PyUnicode_gettyperecord(ch)
+    if (ctype.flags & _PyUnicode_TyperecordMasks.EXTENDED_CASE_MASK):
+        index = ctype.title & 0xFFFF
+        n = ctype.title >> 24
+        for i in range(n):
+            res[i] = _PyUnicode_ExtendedCase(index + i)
+        return n
+    res[0] = ch + ctype.title
+    return 1
 
 
-# From: https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodectype.c#L243-L257
+# From: https://github.com/python/cpython/blob/201c8f79450628241574fba940e08107178dc3a5/Objects/unicodectype.c#L243-L257
```

Comment:
Why is this SHA being changed? The code to which it is referring has not changed in this PR?

Reply:
Let me revert the change.